pkg/cmd/auth/login/login.go

```diff
@@ -138,6 +138,7 @@ func loginRun(opts *LoginOptions) error {
 			return err
 		}
 	}
+	hostname = ghinstance.NormalizeHostname(hostname)
```

Comment:
Hi, thank you for attempting to fix this, but I would rather see that the original cause of the bug was addressed and discovered.

For example, it should be valid to do `gh auth login -h GITHUB.COM` and have the whole authentication flow complete and the token getting stored under `GITHUB.COM`. We should be matching hostnames case-insensitively throughout the codebase without having to normalize hostnames at the time of logging in and storing tokens.

I would appreciate if the bug fix was a change that was applied to the part of the codebase where the error that was originally reported comes from. Thank you!

Reply:
How should the app handle if the user is already logged in to `github.com`, then reauthenticate with `GitHub.com`? Should we save both of them in `hosts.yml`? 🤔

Reply:
Furthermore, the cause of the bug in #6979 is because `cfg.AuthToken(hostname)` method normalized the `hostname` internally. If we want to store and validate the user-input hostname, we'd have to modify this method.

Reply:
Good point about the `cfg.AuthToken` method. I have looked up how that works and saw that the fixes would have to be applied to our external library (go-gh). I've proposed some changes there: https://github.com/cli/go-gh/pull/105

If we decide to go that route and the go-gh change gets merged, there will be no need for this PR, as the issue will be solved. However, I thank you for proposing to fix this in the first place. Let's first see what happens with the other PR; I will update this thread accordingly.